### PR TITLE
Add TLS cert checksum

### PR DESCRIPTION
**Pull request by klauspost**
## Description

Allows to validate if all certs match across cluster.

Requires https://github.com/minio/madmin-go/pull/251

## How to test this PR?

Submit health check.

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -247,3 +247,5 @@ require (
 	gopkg.in/square/go-jose.v2 v2.6.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )
+
+replace github.com/minio/madmin-go/v3 => github.com/klauspost/madmin-go/v3 v3.0.0-20231129155207-e7ce2dbac649
```
Comment (aead) [inline go.mod]: The replace should be removed, right?!

Comment (klauspost) [reply go.mod]: Yeah. Once https://github.com/minio/madmin-go/pull/251 is merged. It is there so CI can compile.

**Review by harshavardhana — APPROVED** [review]

**Review by vadmeste — COMMENTED** [review]
wondering why we don't use the certificate serial number ? just we do not want to be aggressive?

**Comment by harshavardhana** [comment]
> wondering why we don't use the certificate serial number ? just we do not want to be aggressive?

Serial number apparently can be re-used or shared? @vadmeste

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**